### PR TITLE
Search by date worked up and operational

### LINKED ISSUE #22: 12 Search by Date
```
As a user
So I can fund a space that suits me
I want to find one by availability
```

### PR DESCRIPTION
**Pull request by Gleoman**
Closes #22 

Search functionality now fully operational from the listings page and subsequent search return page also.

Allows users to search by date availability.

Perhaps further development could include searching by other parameters, for example name/owner, description/location.

### CODE REVIEW

File: app/app.rb [hunk 1]
```diff
@@ -54,6 +54,15 @@ class DwellBNB < Sinatra::Base
     erb :'spaces/listings'
   end
 
+	post '/spaces/availability' do
+		@listings = []
+		params[:check_availability].split.each do |date|
+			d = Date.strptime(date, '%d/%m/%Y')
```
Comment (alextwilson) [inline app/app.rb]: I think d should be renamed to date or something similar

**Comment by alextwilson** [comment]
:shipit:

**Comment by Gleoman** [comment]
:shipit:

**Comment by Leigan0** [comment]
🕋

**Pull request merged**